### PR TITLE
SPI address line console

### PR DESCRIPTION
**Pull request by rminnich**
uart::spi::SPI implements a uart in which the characters are
output on SPI address lines 13:6. A properly patched em100
program can read this console. This is the console you use when things
are really screwed.

Doesn't even compile.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: src/drivers/uart/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,8 @@ i8250 = ["heapless"]
 amdmmio = []
 pl011 = []
 ns16550 = []
-sifive = []
+spi = []
+ifive = []
```
Comment (rjoleary) [inline src/drivers/uart/Cargo.toml:25]: sifive

**Review by rjoleary — DISMISSED** [review]

**Review by rjoleary — APPROVED** [review]

**Comment by rjoleary** [comment]
Clippy's back!

I don't even think this is in the code you touched:

```
error[E0658]: use of unstable library feature 'proc_macro_is_available'
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.33/src/detection.rs:28:21
   |
28 |     let available = proc_macro::is_available();
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #71436 <https://github.com/rust-lang/rust/issues/71436> for more information
   = help: add `#![feature(proc_macro_is_available)]` to the crate attributes to enable
```

**Comment by orangecms** [comment]
> Clippy's back!
> 
> I don't even think this is in the code you touched:
> 
> ```
> error[E0658]: use of unstable library feature 'proc_macro_is_available'
>   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.33/src/detection.rs:28:21
>    |
> 28 |     let available = proc_macro::is_available();
>    |                     ^^^^^^^^^^^^^^^^^^^^^^^^
>    |
>    = note: see issue #71436 <https://github.com/rust-lang/rust/issues/71436> for more information
>    = help: add `#![feature(proc_macro_is_available)]` to the crate attributes to enable
> ```

This is a build error, not a lint error, isn't it? I'm looking into this. We're on 1.58.0, which has a different feature set.
If in doubt, just add `#![feature(proc_macro_is_available)]` to `src/drivers/uart/src/lib.rs`.

**Comment by orangecms** [comment]
Okay, the toolchain upgrade seems to have fixed the issue - on we go! \o/

**Pull request merged**